### PR TITLE
[CPP, Typescript-Node] - Ubuntu-24.04 (noble) & 22-bookworm - updated

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Template names**:
 
 * `CPP, Typescript-Node`
 
 **Description**:
 
 This PR makes the following changes : 
 1. Added `Ubuntu-24.04` to `CPP's` `devcontainer-template.json`
 2. Added `22-bookworm` to `Typescript-Node's` `devcontainer-template.json`

 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

File: src/cpp/devcontainer-template.json [hunk 1]
```diff
@@ -9,11 +9,12 @@
     "options": {
         "imageVariant": {
             "type": "string",
-            "description": "Debian / Ubuntu version (use Debian 12, Debian 11, Ubuntu 22.04 on local arm64/Apple Silicon):",
+            "description": "Debian / Ubuntu version (use Debian 12, Debian 11, Ubuntu 24.04 on local arm64/Apple Silicon):",
```
Comment (samruddhikhandale) [inline src/cpp/devcontainer-template.json:12]:
```suggestion
            "description": "Debian / Ubuntu version (use Debian 12, Debian 11, Ubuntu 24.04, Ubuntu 22.04 on local arm64/Apple Silicon):",
```

File: src/typescript-node/devcontainer-template.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
             "type": "string",
             "description": "Node.js version (use -bookworm, -bullseye variants on local arm64/Apple Silicon):",
             "proposals": [
+				"22-bookworm",
```
Comment (samruddhikhandale) [inline src/typescript-node/devcontainer-template.json:14]: can we add entries for 22-bullseye?

Comment (samruddhikhandale) [inline src/typescript-node/devcontainer-template.json:14]: Also, let's set `22-bookworm` as default in L22

File: src/cpp-mariadb/devcontainer-template.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
 				"debian-12",
                 "debian-11",
                 "debian-10",
+				"ubuntu-24.04",
```
Comment (samruddhikhandale) [inline src/cpp-mariadb/devcontainer-template.json:17]: Can we update the description for this Template as well?

File: test/typescript-node/eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
Comment (samruddhikhandale) [inline test/typescript-node/eslint.config.js:1]: eslint test is still failing, can you correct the test by referring to my PR which added node 22 support to TS-node?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/cpp/devcontainer-template.json (hunk 1 above)
Comment (gauravsaini04) [reply src/cpp/devcontainer-template.json]: done

File: src/typescript-node/devcontainer-template.json (hunk 2 above)
Comment (gauravsaini04) [reply src/typescript-node/devcontainer-template.json:14]: done

File: src/cpp-mariadb/devcontainer-template.json (hunk 3 above)
Comment (gauravsaini04) [reply src/cpp-mariadb/devcontainer-template.json:17]: done

File: src/typescript-node/devcontainer-template.json (hunk 2 above)
Comment (gauravsaini04) [reply src/typescript-node/devcontainer-template.json:14]: done

**Review by samruddhikhandale — APPROVED** [review]

**Pull request merged**